convert2rhel/pkghandler.py

```diff
@@ -433,26 +448,28 @@ def install_rhel_kernel():
     loggerinst.info("Installing RHEL kernel ...")
     output, ret_code = call_yum_cmd(command="install", args="kernel")
 
-    # check condition - failed installation
-    if ret_code:
+    if ret_code != 0:
         loggerinst.critical("Error occured while attempting to install the"
                             " RHEL kernel")
 
-    # check condition - kernel with same version is already installed
-    already_installed = re.search(r" (.*?) already installed", output)
+    # Check if kernel with same version is already installed.
+    # Example output from yum and dnf:
+    #  "Package kernel-2.6.32-754.33.1.el6.x86_64 already installed and latest version"
+    #  "Package kernel-4.18.0-193.el8.x86_64 is already installed."
+    already_installed = re.search(r" (.*?)(?: is)? already installed", output, re.MULTILINE)
     if already_installed:
-        kernel_version = already_installed.group(1)
-        kernel_obj = get_installed_pkgs_w_different_fingerprint(
-            system_info.fingerprints_rhel, kernel_version)
-        if kernel_obj:
-            # If the installed kernel is from a third party (non-RHEL) and has
-            # the same NEVRA as the one available from RHEL repos, it's
-            # necessary to install an older version RHEL kernel, because the
-            # third party one will be removed later in the conversion process.
-            loggerinst.info("\nConflict of kernels: One of the installed kernels"
-                            " has the same version as the latest RHEL kernel.")
-            handle_no_newer_rhel_kernel_available()
-            return True
+        rhel_kernel_nevra = already_installed.group(1)
+        non_rhel_kernels = get_installed_pkgs_w_different_fingerprint(system_info.fingerprints_rhel, "kernel")
```

Comment:
Previously we sent in `already_installed.group(1)` as second argument. Why are we sending in `"kernel"` now?

Reply:
Thanks for the question. It's related to what I wrote in the description: "_The get_installed_pkgs_w_different_fingerprint function, when using DNF, is filter by package name only, not by package NEVRA as with YUM._" I'll try my best to explain it further.

Before supporting conversions to RHEL 8, when we were using yum only, we could get a yum package object by filtering the all the package objects by NEVRA. So for example, we passed "`kernel-2.6.32-754.33.1.el6.x86_64`" to this [yum API function](https://github.com/oamg/convert2rhel/blob/master/convert2rhel/pkghandler.py#L230) and it gave us the object of that installed kernel package.

Now when using dnf, we've lost the possibility to filter packages by NEVRA, at least from what I've found. We can filter the packages by name, epoch, version, release or architecture, but separately. So with dnf, I'm [filtering the packages by name](https://github.com/oamg/convert2rhel/blob/master/convert2rhel/pkghandler.py#L241).

That's why there's just "`kernel`" - to get a list of all the installed kernels. Because I can't filter by NEVRA anymore.